scipy/spatial/qhull.pyx

```diff
@@ -2410,3 +2449,269 @@ class Voronoi(_QhullUser):
+        points are input points which were *not* included in the
+        triangulation due to numerical precision issues.
+
+        If option "Qc" is not specified, this list is not computed.
+    dual_area : float
+        Area of the dual convex hull
+    dual_volume : float
+        Volume of the dual convex hull
+
+    Raises
+    ------
+    QhullError
+        Raised when Qhull encounters an error condition, such as
+        geometrical degeneracy when options to resolve are not enabled.
+    ValueError
+        Raised if an incompatible array is given as input.
+
+    Notes
+    -----
+    The intersections are computed using the 
+    `Qhull library <http://www.qhull.org/>`__.
+
+    Examples
+    --------
+
+    TODO: example
+    Halfspace intersection of planes forming some polygon
+
+    >>> from scipy.spatial import HalfspaceIntersection
+    >>> import numpy as np
+    >>> halfspaces = np.array([[-1, 0., 0.],
+    >>>                        [0., -1., 0.],
+    >>>                        [2., 1., -4.],
+    >>>                        [-0.5, 1., -2.]])
+    >>> feasible_point = np.array([0.5, 0.5])
+    >>> hs = HalfspaceIntersection(halfspaces, feasible_point)
+
+    Plot halsfpaces as filled regions and intersection points:
+
+    >>> import matplotlib.pyplot as plt
+    >>> xlim, ylim = (-1, 3), (-2, 6)
+    >>> x = np.linspace(-1, 3, 100)
+    >>> symbols = ['-', '+', 'x', '*']
+    >>> signs = [0, 0, -1, -1]
+    >>> for h, sym, sign in zip(halfspaces, symbols, signs):
+    >>>   if h.item(1) == 0:
+    >>>     plt.axvline(-h.item(2)/h.item(0), label='{}x+{}y+{}=0'.format(*h.tolist()))
+    >>>     xi = np.linspace(xlim[sign], -h.item(2)/h.item(0), 100)
+    >>>     plt.fill_between(xi, ylim[0], ylim[1], color="none", edgecolor="b", hatch=sym, alpha=0.5)
+    >>>   else:
+    >>>     plt.plot(x, (-h.item(2)-h.item(0)*x)/h.item(1), label='{}x+{}y+{}=0'.format(*h.tolist()))
+    >>>     plt.fill_between(x, (-h.item(2)-h.item(0)*x)/h.item(1), ylim[sign], color="none", edgecolor="b", hatch=sym, alpha=0.5)
+    >>> x, y = zip(*hs.intersections)
+    >>> plt.plot(x, y, 'o', markersize=8)
+    >>> plt.legend()
+    >>> plt.show()
+
+    References
+    ----------
+    .. [Qhull] http://www.qhull.org/
+
+    """
+
+    def __init__(self, halfspaces, feasible_point,
+                    incremental=False, qhull_options=None):
+        if np.ma.isMaskedArray(halfspaces):
+            raise ValueError('Input halfspaces cannot be a masked array')
+        if np.ma.isMaskedArray(feasible_point):
+            raise ValueError('Input feasible point cannot be a masked array')
+        if feasible_point.shape[-1] != halfspaces.shape[1]-1:
+            raise ValueError('The last dimension of feasible point is not dimension of halfspaces minus one')
+        halfspaces = np.ascontiguousarray(halfspaces, dtype=np.double)
+        self.feasible_point = np.ascontiguousarray(feasible_point, dtype=np.double)
```

Comment:
feasible_point probably should be 1d array, not shape (1, ndim).
The ndim should be also checked.